textdb/textdb-textql/src/test/java/edu/uci/ics/textdb/textql/statements/predicates/KeywordExtractPredicateTest.java

```diff
@@ -0,0 +1,162 @@
+        
+        // Check the constructor with the arguments for initialization
+        keywordExtractPredicate = new KeywordExtractPredicate(matchingFields, keywords, matchingType);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingFields(), matchingFields);
+        Assert.assertEquals(keywordExtractPredicate.getKeywords(), keywords);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingType(), matchingType);
+        
+        // Check the constructor with no arguments
+        keywordExtractPredicate = new KeywordExtractPredicate();
+        Assert.assertEquals(keywordExtractPredicate.getMatchingFields(), null);
+        Assert.assertEquals(keywordExtractPredicate.getKeywords(), null);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingType(), null);
+        
+        // Check all the setters with the given values
+        keywordExtractPredicate.setMatchingFields(matchingFields);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingFields(), matchingFields);
+        keywordExtractPredicate.setKeywords(keywords);
+        Assert.assertEquals(keywordExtractPredicate.getKeywords(), keywords);
+        keywordExtractPredicate.setMatchingType(matchingType);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingType(), matchingType);
+        
+        // Check all the setters with null values
+        keywordExtractPredicate.setMatchingFields(null);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingFields(), null);
+        keywordExtractPredicate.setKeywords(null);
+        Assert.assertEquals(keywordExtractPredicate.getKeywords(), null);
+        keywordExtractPredicate.setMatchingType(null);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingType(), null);
+    }
+
+    /**
+     * Test the getOperatorBean method.
+     * Build a KeywordExtractPredicate, invoke the getOperatorBean and
+     * check whether a KeywordMatcherBean with the right attributes is returned.
+     * An empty list is used as the list of fields to perform the match.
+     */
+    @Test
+    public void testGetOperatorBean00() {
+        List<String> matchingFields = Collections.emptyList();
+        String keywords = "keyword";
+        String matchingType = KeywordMatchingType.CONJUNCTION_INDEXBASED.toString();
+        KeywordExtractPredicate keywordExtractPredicate = new KeywordExtractPredicate(matchingFields, keywords, matchingType);
+        
+        OperatorBean computedProjectionBean = keywordExtractPredicate.getOperatorBean("xxx");
+        String matchingFieldsAsString = String.join(",", matchingFields);
+        OperatorBean expectedProjectionBean = new KeywordMatcherBean("xxx", "KeywordMatcher", matchingFieldsAsString,
+                            null, null, keywords, matchingType);
+        
+        Assert.assertEquals(expectedProjectionBean, computedProjectionBean);
+    }
+    
+    /**
+     * Test the getOperatorBean method.
```

Comment:
See comments above. In the new PR, rename "getOperatorBean" to "generateOperatorBean".